crates/krilla/src/object/embed.rs

```diff
@@ -0,0 +1,181 @@
+//! Embedding attachments to a PDF file.
+
+use crate::metadata::pdf_date;
+use crate::object::{Cacheable, ChunkContainerFn};
+use crate::serialize::SerializeContext;
+use crate::stream::FilterStreamBuilder;
+use crate::util::NameExt;
+use crate::validation::Validator;
+use crate::version::PdfVersion;
+use pdf_writer::types::AssociationKind;
+use pdf_writer::{Chunk, Finish, Name, Ref, Str, TextStr};
+use std::ops::DerefMut;
+use std::sync::Arc;
+
+/// An error while embedding the file.
+pub enum EmbedError {
+    /// The document doesn't contain a date, which is required for embedded files
+    /// in some export modes.
+    MissingDate,
+}
+
+/// An embedded file.
+#[derive(Debug, Clone, Hash)]
+pub struct EmbeddedFile {
+    /// The name of the embedded file.
+    pub path: String,
+    /// The mime type of the embedded file.
+    pub mime_type: String,
+    /// A description of the embedded file.
+    pub description: Option<String>,
+    /// The association kind of the embedded file.
+    pub association_kind: AssociationKind,
+    /// The raw data of the embedded file.
+    pub data: Arc<Vec<u8>>,
```

Comment:
This is more of a general comment: It would be nice if krilla would take a more generic type like `Arc<dyn AsRef<[u8]> + Send + Sync>` or some custom container like that, so that one doesn't have to copy the data if the source representation differs a bit.